UI/obs-app.cpp

```diff
@@ -768,7 +768,12 @@ bool OBSApp::MigrateGlobalSettings()
 		return false;
 	}
 
-	std::filesystem::copy(legacyGlobalConfigFile, userConfigFile);
+	try {
+		std::filesystem::copy(legacyGlobalConfigFile, userConfigFile);
+	} catch (const std::filesystem::filesystem_error &error) {
+		blog(LOG_ERROR, "Unable to migrate global configuration - copy failed.\n%s", error.what());
```

Comment:
Use an `OBSErrorBox` just like all the other prior error cases.

Reply:
I followed the two existing examples in the function (`move_basic_to_profiles`) which are
```
        try {
                std::filesystem::create_directories(profilesPath);
        } catch (const std::filesystem::filesystem_error &error) {
                blog(LOG_ERROR, "Failed to create profiles directory for migration from basic profile\n%s",
                     error.what());
                return;
        }
```
and
```
                if (!std::filesystem::exists(newProfilePath)) {
                        try {
                                std::filesystem::create_directory(newProfilePath);
                        } catch (const std::filesystem::filesystem_error &error) {
                                blog(LOG_ERROR, "Failed to create profile directory for 'Untitled'\n%s", error.what());
                                return;
                        }
                }
```

Reply:
In some cases the code just prints an error, but in other cases program execution needs to abort immediately because it creates an unrecoverable state.

Any error in `MigrateGlobalSettings` is considered unrecoverable, hence why all error conditions use `OBSErrorBox` in this function and this error condition is not different.

Reply:
Shall I change this to fix the two existing cases in `move_basic_to_profiles`?

Reply:
Hang on, looking at wrong branch :(